packages/namekit-react/src/components/DisplayedPriceWithSymbol.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  Currency,
+  CurrencySymbology,
+  PriceCurrencyFormat,
+  getCurrencySymbology,
+} from "@namehash/ens-utils";
+import React from "react";
+import {
+  CurrencySymbol,
+  CurrencySymbolSize,
+} from "./CurrencySymbol/CurrencySymbol";
+import cc from "classcat";
+import { Tooltip } from "./Tooltip";
+import {
+  DisplayedPrice,
+  PriceDisplaySize,
+  DisplayedPriceProps,
+} from "./DisplayedPrice";
+
+export const CurrencySymbolPosition = {
+  /**
+   * Display the currency symbol to the left of the price. (e.g. $1.00)
+   */
+  Left: "nk-mr-1.5",
+  /**
+   * Display the currency symbol to the right of the price. (e.g. 1.00$)
+   */
+  Right: "nk-ml-1.5",
+} as const;
+export type CurrencySymbolPosition =
+  (typeof CurrencySymbolPosition)[keyof typeof CurrencySymbolPosition];
+
+export interface DisplayedPriceWithSymbolProps extends DisplayedPriceProps {
+  // Wether to display a currency symbology as its acronym or symbol
+  currencySymbology?: CurrencySymbology;
+  /**
+   * The position of the currency symbol relative to the price.
+   * Defaults to `CurrencySymbolPosition.Left`.
+   *
+   * If `CurrencySymbolPosition.Left`, the currency symbol will be displayed to the left of the price.
+   * If `CurrencySymbolPosition.Right`, the currency symbol will be displayed to the right of the price.
+   */
+  symbolPosition?: CurrencySymbolPosition;
+  // Wether or not to display the name of the currency in a tooltip when its symbol is hovered
+  describeCurrencyInTooltip?: boolean;
+}
+
+/**
+ * A mapping from a `PriceDisplaySize` to a `CurrencySymbolSize`.
+ * This is useful for determining the size of the currency symbol to display
+ * for a given price display size. e.g. as needed in DisplayedPrice, when a
+ * currency symbol is displayed alongside a price and the user can define
+ * the size of the price display. This way both price display size and
+ * currency symbol size can be controlled by the same prop, always
+ * guaranteeing that they are displayed with aligned sizes.
+ * @param priceDisplaySize: PriceDisplaySize. The size of the price display.
+ * @returns CurrencySymbolSize. The size of the currency symbol to display.
+ */
+export const getCurrencySymbolSize = (
+  priceDisplaySize: PriceDisplaySize,
+): CurrencySymbolSize => {
+  switch (priceDisplaySize) {
+    case PriceDisplaySize.Micro:
+    case PriceDisplaySize.Small:
+      return CurrencySymbolSize.Small;
+    case PriceDisplaySize.Medium:
+    case PriceDisplaySize.Large:
+      return CurrencySymbolSize.Large;
+  }
+};
+
+export const DisplayedPriceWithSymbol = ({
```

Comment:
I don't understand why this UI component should exist. In my mind it shouldn't -- or if it does it should be an extremely thin wrapper around other components.

Reply:
@FrancoAguzzi In other words, there's way too much logic in this UI component. That's a strong sign that we aren't writing our code in a properly composable way. If you get the composition correct than there should be very little logic in this file.

Reply:
@FrancoAguzzi Very important: you need to think more about the context where these UI components are used in an overall app. You want to be giving power to that context and stop trying to control everything inside these UI components you're defining directly.

Reply:
Hi @lightwalker-eth !

This component is a wrapper around DisplayedPrice that gives us the ability to render a price with:

1. Left aligned symbology
2. Right aligned symbology
3. Acronym symbology for inline DisplayedPrice 
4. Icon (Symbol) symbology for inline DisplayedPrice
5. Acronym symbology for _Tooltiped_ DisplayedPrice 
6. Icon (Symbol) symbology for _Tooltiped_ DisplayedPrice

The implementation is pretty concise, maybe what you are aiming here is not to achieve a smaller comp. definition but a comp. definition with fewer variants?